### PR TITLE
INT-378_single_node_optional_license

### PR DESCRIPTION
**Pull request by fernau**
JIRA: https://issues.sonatype.org/secure/RapidBoard.jspa?rapidView=123
Jenkins integration test: https://jenkins.zion.aws.s/job/integrations/job/nexus-ha-reference-feature/job/INT-378_single_node_optional_license/

### CODE REVIEW

File: recipes/aws.rb [hunk 1]
```diff
@@ -103,15 +103,16 @@
...
 python 'check instance health' do
-  code <<-EOH
-import boto3, sys
-instance_id = open('/var/lib/cloud/data/instance-id', 'r').readline().strip()
-elb_client = boto3.client('elb', region_name='#{region}')
-elbs = elb_client.describe_load_balancers()['LoadBalancerDescriptions']
-elb_name = [e['LoadBalancerName'] for e in elbs if instance_id in [ i['InstanceId'] for i in e['Instances'] ]]
-inst_health = elb_client.describe_instance_health(LoadBalancerName=elb_name[0], Instances=[{'InstanceId':instance_id}])
-if inst_health['InstanceStates'][0]['State'] != 'InService': sys.exit(1)
+  code <<-EOH.gsub(/^\s+/, '')
```
Comment (bigspotteddog) [inline recipes/aws.rb]: 👍

File: templates/nexus.properties.erb [hunk 2]
```diff
@@ -13,3 +13,6 @@ nexus.clustered=<%= node['nexus_repository_manager']['properties']['clustered']
 <% if node['nexus_repository_manager']['properties']['license_file'] %>
 nexus.licenseFile=<%= node['nexus_repository_manager']['properties']['license_file'] %>
 <% end %>
+<% if node['nexus_repository_manager']['license_s3_bucket'] != '' %>
```
Comment (bigspotteddog) [inline templates/nexus.properties.erb]: I think this should be an else if condition to the license_file condition above.

File: recipes/download.rb [hunk 3]
```diff
@@ -92,3 +92,13 @@
   owner 'root'
   group 'root'
 end
+
+s3_file node['nexus_repository_manager']['nexus_data']['etc']['path'] + '/license.lic' do
```
Comment (whyjustin) [inline recipes/download.rb]: Where was this placed before we made it conditional?

File: templates/nexus.properties.erb (hunk 2 above)
Comment (whyjustin) [reply templates/nexus.properties.erb]: +1

File: recipes/download.rb (hunk 3 above)
Comment (fernau) [reply recipes/download.rb]: this is now conditional, see the 'not_if' condition a few lines down.

File: templates/nexus.properties.erb (hunk 2 above)
Comment (fernau) [reply templates/nexus.properties.erb]: created 'elsif' (not a typo) in 6870a11

**Pull request merged**